masonry/src/text/edit.rs

```diff
@@ -364,49 +334,39 @@ impl<T: EditableText> TextEditor<T> {
             active_affinity: Affinity::Downstream,
             h_pos: None,
         });
-        self.text_mut().edit(selection.range(), &**c);
+        self.text_mut().replace_range(selection.range(), c);
         self.inner.selection = Some(Selection::caret(
             selection.min() + c.len(),
             // We have just added this character, so we are "affined" with it
             Affinity::Downstream,
         ));
-        let contents = self.text().as_str().to_string();
+        let contents = self.text().clone();
         ctx.submit_action(Action::TextChanged(contents));
         Handled::Yes
     }
 }
 
-impl<T: EditableText> Deref for TextEditor<T> {
-    type Target = TextWithSelection<T>;
+impl Deref for TextEditor {
+    type Target = TextWithSelection<String>;
```

Comment:
I suspect we might want to remove this Deref at some point - it's a bit messy. Not a job for this PR